openedx/core/lib/hash_utils.py

```diff
@@ -32,3 +35,32 @@ def short_token():
     Currently, this is just meant to create sufficiently random tokens
     """
     return create_hash256(max_length=32)
+
+
+def hash_usage_key(usage_key: UsageKey) -> str:
+    """
+    Get the blake2b hash key for the given usage_key and encode the value. The
+    hash key will be added to the usage key's mapping dictionary for decoding
+    in LMS.
+
+    Args:
+        usage_key: the id of the location to which to generate the path
+
+    Returns:
+        The string of the encoded hashed key.
+    """
+    short_key = hashlib.blake2b(bytes(str(usage_key), 'utf-8'), digest_size=6)
+    encoded_hash = urlsafe_b64encode(bytes(short_key.hexdigest(), 'utf-8'))
+    return str(encoded_hash, 'utf-8')
+
+
+def is_potential_usage_key_hash(string: str) -> bool:
+    """
+    Is this string a possible output of `hash_usage_key`?
+    """
+    lowers = {chr(i) for i in range(ord('a'), ord('z') + 1)}
+    uppers = {chr(i) for i in range(ord('A'), ord('Z') + 1)}
+    digits = {chr(i) for i in range(ord('0'), ord('9') + 1)}
+    symbols = {"-", "_"}
+    all_hash_characters = lowers | uppers | digits | symbols
+    return set(string).issubset(all_hash_characters)
```

Comment:
I had written this to facilitate better error reporting (ie "invalid sequence identifier" vs. "sequence not found"), but it's not being used in this draft of the PR. I may end up rmoving it.

Reply:
I ended up removing it in favor of a regex, which one can use to check if something's a usage key hash.